### PR TITLE
fix(Reply): Add check if responsed message was deleted and add `<deleted>` to reply message

### LINKED ISSUE #7372: reply to deleted message
### Description

1. send a message
2. reply to message
3. delete original message

result: original message still shows in the reply
expected: show `<deleted>` in the reply

### PR DESCRIPTION
**Pull request by borismelnik**
Closes: #7372

### What does the PR do

Add methods to `message_model` to for update and find messages which contains reply with deleted messages.

### Affected areas

Messages

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [ ] I've checked the design and this PR matches it

Снимок экрана 2022-10-06 в 16 51 51

### CODE REVIEW

File: src/app/modules/shared_models/message_model.nim [hunk 1]
```diff
@@ -82,6 +83,12 @@ QtObject:
   method rowCount(self: Model, index: QModelIndex = nil): int =
     return self.items.len
 
+  method isMessageExists(self: Model, messageId: string): bool =
```
Comment (caybro) [inline src/app/modules/shared_models/message_model.nim:86]: I'd suggest better wording, sth like `doesMessageExist()`

File: ui/imports/shared/views/chat/MessageView.qml [hunk 2]
```diff
@@ -587,7 +589,8 @@ Loader {
             }
 
             replyDetails: StatusMessageDetails {
-                messageText:  delegate.replyMessage ? delegate.replyMessage.messageText : ""
+                messageText:  delegate.replyMessage ? delegate.replyMessage.messageText
+                                                    : qsTr("&lt;deleted&gt;")
```
Comment (caybro) [inline ui/imports/shared/views/chat/MessageView.qml:593]: the `qsTr()` needs some context; "deleted" is an adjective but the translator wouldn't know what the subject is

Comment (borismelnik) [reply ui/imports/shared/views/chat/MessageView.qml:593]: why? What do you mean? any example?

Comment (caybro) [reply ui/imports/shared/views/chat/MessageView.qml:593]: https://doc.qt.io/qt-5/qtquick-internationalization.html#2-add-context-for-the-translator

**Review by MishkaRogachev — APPROVED** [review]

File: ui/imports/shared/views/chat/MessageView.qml (hunk 2 above)
Comment (caybro) [reply ui/imports/shared/views/chat/MessageView.qml:594]: "deleted message" would've been sufficient, but thanks :)

**Review by caybro — APPROVED** [review]

**Review by igor-sirotin — APPROVED** [review]

**Pull request merged**